### ISSUE TITLE
Duplicate draft release created when published release with same name exists

### ISSUE THREAD

**Issue by daviwil** [remixz/publish-release#37]
Hi there!  I'm trying to use this library to automatically create GitHub releases and upload assets as a CI step.  I've noticed that when `reuseRelease` and `reuseDraftOnly` are true, a duplicate draft release gets created if there's already a published release by the same name.  

Here's what I'm trying to accomplish:

- Create a draft release for version v1.0.0 if one doesn't already exist and there's no published release by the same name
- If a draft v1.0.0 release exists, update the release and replace its assets
- If a published v1.0.0 release exists, don't create a new draft release for v1.0.0 and don't update existing assets

It doesn't seem possible to do this currently since using `reuseRelease` and `reuseDraftOnly` causes a new draft release to be created if one doesn't already exist with the same name.

Is this the intended behavior?  If not, could I send a PR for a new parameter that will skip draft release creation if there's already a published release with the same name?

Thanks!

**Comment by daviwil** [comment]
One thought I had for the name of such an option is `skipIfPublished`.  Might try prototyping that and send a PR.

**Comment by marceloavf** [comment]
We could disable a `draft release` for v1.0.0 when there is a `release` v1.0.0 since you will never be able to publish this specific draft release because you can't duplicate tags in a final release/pre release. 

But I'm thinking about the users that use the `editRelease` option to update this draft release tag to a new one before publishing it.

**Comment by daviwil** [comment]
Yeah, that's why I think it's better for there to be a parameter to explicitly opt in to the new behavior.

I've created https://github.com/remixz/publish-release/pull/38 to add a new parameter for this, let me know what you think!
